internal/service/user.go

```diff
@@ -20,35 +20,55 @@ func NewFileService(repo repository.UserRepository) FileService {
 	return &fileService{repo: repo}
 }
 
-func (r *fileService) HandlerFile(fileName string) {
+func (r *fileService) HandlerFile(fileName string) error {
 	file, err := os.Open(fileName)
 	if err != nil {
-		log.Fatalf("Cannot open '%s': %s\n", fileName, err.Error())
+		return &domain.Error{
+			NameFile: fileName,
+			Err:      err,
+			Msg:      "Cannot open file",
+		}
 	}
 	defer file.Close()
-	rows := ReadFile(file)
+	rows, err := ReadFile(file)
+	if err != nil {
+		return &domain.Error{
+			NameFile: fileName,
+			Err:      err,
+			Msg:      "Error al leer el archivo",
+		}
+	}
 	for _, row := range rows {
 		user, err := domain.RowFileToUser(row)
 		if err != nil {
-			fmt.Printf("Error ID format: %+v\n", row)
+			e := domain.Error{
+				NameFile: fileName,
+				Err:      err,
+				Msg:      fmt.Sprintf("Error ID format: %+v\n", row),
+			}
+			log.Println(e.Error())
 			continue
 		}
 		err = r.repo.Create(user)
 		if err != nil {
-			log.Println(err)
+			e := domain.Error{
+				NameFile: fileName,
+				Err:      err,
+				Msg:      "Error creating user",
+			}
+			log.Println(e.Error())
 		}
 	}
-	fmt.Printf("Archivo %s cargado\n", fileName)
+	return nil
 }
 
-func ReadFile(file *os.File) [][]string {
+func ReadFile(file *os.File) ([][]string, error) {
```

Comment:
Esto estaría bueno que devuelva un puntero asi evitamos que se duplique en memoria
```suggestion
func ReadFile(file *os.File) (*[][]string, error) {
```

Reply:
Corregido en b8e8f7b